### PR TITLE
Backend implementation of update transaction

### PR DESCRIPTION
**Pull request by GilgameshTC**

### CODE REVIEW

File: bacon/model/StorageCouchBaseDB.swift [hunk 1]
```diff
@@ -97,10 +97,11 @@ class StorageCouchBaseDB {
         return folderPath
     }
 
-    private func createMutableDocument(from transaction: Transaction) throws -> MutableDocument {
+    private func createMutableDocument(from transaction: Transaction, id: String? = nil) throws -> MutableDocument {
         do {
             let transactionData = try transaction.asDictionary()
-            let transactionDocument = MutableDocument(data: transactionData)
+            // If id is nil, a random UID will be assigned to the document
```
Comment (fterh) [inline bacon/model/StorageCouchBaseDB.swift:103]: I thought waikay mentioned that the ID should not be generated by the DB?

edit: Oh, just realized this is a private method. The ID is for internal use only, right?

File: bacon/model/StorageCouchBaseDB.swift [hunk 2]
```diff
@@ -211,6 +212,40 @@ class StorageCouchBaseDB {
         }
     }
 
+    func updateTransaction(_ transaction: Transaction) throws {
+        // Fetch the specific document from database
+        guard let transactionId = transactionMapping[transaction] else {
+            log.info("""
```
Comment (fterh) [inline bacon/model/StorageCouchBaseDB.swift:218]: This should be log.warn (or maybe even error?)

File: bacon/model/StorageCouchBaseDB.swift [hunk 3]
```diff
@@ -211,6 +212,40 @@ class StorageCouchBaseDB {
...
+        let transactionDocument = try createMutableDocument(from: transaction, id: transactionId)
+        log.info("""
+            StorageCouchBaseDB.updateTransaction() with argument:
+            transaction:\(transaction).
+            """)
+        // Update the document
+        do {
+            try transactionDatabase.saveDocument(transactionDocument)
+        } catch {
+            log.info("""
```
Comment (fterh) [inline bacon/model/StorageCouchBaseDB.swift:237]: Same, should be a higher severity log message

**Review by fterh — CHANGES_REQUESTED** [review]

File: bacon/model/StorageCouchBaseDB.swift (hunk 1 above)
Comment (GilgameshTC) [reply bacon/model/StorageCouchBaseDB.swift:103]: ya

**Pull request closed without merging**